### ISSUE TITLE
GetConnectionString() not working when migrated to 2.4.2

### ISSUE THREAD

**Issue by sajanep** [SteeltoeOSS/steeltoe#260]
## Describe the bug
We have the below code in our application.

`var mystring = Configuration.GetConnectionString("myKey");  `

Here Configuration is of type IConfiguration. But after migrating to Steeltoe 2.4.2 versions, it stopped working. Somehow GetConnectionString() is not able to get the values from PCF config server.  Our Program.cs looks like below

```
public class Program
    {
       
        public static void Main(string[] args)
        {
            CreateWebHostBuilder(args).Build().Run();
        }

        public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
            WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .ConfigureAppConfiguration((hostContext, configApp) =>
                {
                    configApp.AddCloudFoundry();
                    configApp.AddConfigServer();
                })
                .ConfigureLogging((builderContext, loggingBuilder) =>
                 {
                     // Add Serilog Dynamic Logger 
                     loggingBuilder.ClearProviders();
                     loggingBuilder.AddSerilogDynamicConsole();
                 })
                .AddCloudFoundryActuators(MediaTypeVersion.V2,
                    ActuatorContext.ActuatorAndCloudFoundry);
    }
}
```
Steeltoe versions are given below

```
<PropertyGroup>
    <TargetFramework>netcoreapp2.2</TargetFramework>
    <GenerateDocumentationFile>true</GenerateDocumentationFile>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.App" />
    <PackageReference Include="Microsoft.AspNetCore.Mvc.Versioning.ApiExplorer" Version="3.2.0" />
    <PackageReference Include="Swashbuckle.AspNetCore" Version="4.0.1" />
    <PackageReference Include="Microsoft.AspNetCore.Mvc.Versioning" Version="3.1.4" />
    <PackageReference Include="Steeltoe.Management.CloudFoundryCore" Version="2.4.2" />
    <PackageReference Include="Steeltoe.Extensions.Configuration.CloudFoundryCore" Version="2.4.2" />
    <PackageReference Include="Steeltoe.Extensions.Configuration.ConfigServerCore" Version="2.4.2" />
  </ItemGroup>
```

## Expected behavior
Should be able to retrieve connection strings via IConfiguration.GetConnectionString().

## Environment (please complete the following information):
 - Platform: PCF 
 - .NET Version : .NET Core 2.2
 - Steeltoe Version : 2.4.2

**Comment by TimHess** [comment]
What was the previous version of Steeltoe in use?

Two items to note:
1.  `.AddConfigServer();` will AddCloudFoundry() automatically, you may remove `configApp.AddCloudFoundry();`
1. .NET Core version 2.2 reached end of life back in December, and v3.0 this week... You should consider either updating to 3.1 or rolling back to 2.1; both are LTS versions.

**Comment by sajanep** [comment]
It was 2.4.0 before

**Comment by TimHess** [comment]
We only had a small handful of changes to the config server client code in that time frame. This is probably not the problem, but I usually see the casing as `ConnectionStrings` ~ there is a change that addresses some case sensitivity issues that's available in our dev feed if you want to give that a try.

Are other values coming back from the config server?
Are you able to get that value out of your `IConfiguration` without using `GetConnectionString()`?

Have you tried looking at the output of the env actuator? There might be some clues about what's happening in there.

**Comment by sajanep** [comment]
Yes. I am able to get other values from config server if I respect the case. E.g. `var myConstring = Configuration.Get("connectionstrings:myKey");` As per microsoft https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1#keys, configuration keys are case insensitive. Looks like Steeltoe has broken this.

**Comment by TimHess** [comment]
Hey, sorry for the delay in getting back to you, I've been on vacation...

I was finally able to test this out, and this was fixed by #247
